### ISSUE TITLE
Request: Support for namespaces

### ISSUE THREAD

**Issue by LaraSQP** [mrlacey/CommentLinks#41]
```
// Link: namespace1.filename.cs
// Link: namespace2.filename.cs
```

The objective is to make it possible for the extension to disambiguate files that have the same filename but are in different namespaces.

The alternative now is to use absolute paths (for some reason relative paths do not work for me, using v2.9.49 in vs2022).

Much appreciated. This is quite a good extension.

**Comment by mrlacey** [comment]
how would it be possible to disambiguate between `namespace1.filename.cs` and `file.name.cs`?

It sounds like the bigger issue here is to fix relative paths not working correctly.

**Comment by LaraSQP** [comment]
Fair point.

Namespaces would seem to require additional notation. Perhaps one of the following:

```
// Link: [namespace1]filename.cs
// Link: namespace2>filename.cs
// Link: namespace2+filename.cs
```

But I can see the likely-unintuitive complication and, while these characters are invalid in paths, they are not in URLs.

Thank you for considering the matter.

**Comment by mrlacey** [comment]
Linking by namespace also produces other possibly challenging scenarios in that the namespace doesn't have to match the directory structure and a file can potentially contain multiple namespaces. 
This would lead to the extension needing to be able to understand and pass the code to know about namespaces.
And, as this extension is language and file type agnostic it would either need to know how to parse every language that supports namespaces. Or, it would need to limit functionality by file type/extension.

I think fixing the relative path issue is a better way to go. ;)

**Comment by LaraSQP** [comment]
Makes sense.

ta

**Comment by mrlacey** [comment]
I've just released v2.10 to the marketplace and this includes improvements to relative navigation.
Let me know if you still have issues after updating to this version.

**Comment by LaraSQP** [comment]
Relative paths work.

Much appreciated.
